### ISSUE TITLE
[1.12] Dedicated Server Crash

### ISSUE THREAD

**Issue by Burgersim** [Xilef11/runesofwizardry-classics#8]
Hope It's ok, thought just for a better overview I'd post this in the right place, so to speak, sorry if that seems rude or something like that, I apologize in advance. Using an up-to-date version of Runes of Wizardry, my dedicated Server still crashes with this crash report:

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  LoadingPlugin (Quark-r1.2-104.jar)
  IELoadingPlugin (ImmersiveEngineering-0.12-70.jar)
  IvToolkit (IvToolkit-1.3.3-1.12.jar)
  Extra Alchemy ASM (extraalchemy-0.3.4.1-release.jar)
Contact their authors BEFORE contacting forge

// I let you down. Sorry :(

Time: 8/9/17 7:59 PM
Description: Exception in server tick loop

net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Runes of Wizardry - Classic Dusts Pack (runesofwizardry_classics)
Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/util/ITooltipFlag
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetPublicMethods(Class.java:2902)
	at java.lang.Class.privateGetPublicMethods(Class.java:2917)
	at java.lang.Class.privateGetPublicMethods(Class.java:2917)
	at java.lang.Class.privateGetPublicMethods(Class.java:2917)
	at java.lang.Class.getMethods(Class.java:1615)
	at net.minecraftforge.fml.common.eventhandler.EventBus.register(EventBus.java:81)
	at xilef11.mc.runesofwizardry_classics.inscriptions.InscriptionLeapII.<init>(InscriptionLeapII.java:23)
	at xilef11.mc.runesofwizardry_classics.ModInscriptions.initInscriptions(ModInscriptions.java:39)
	at xilef11.mc.runesofwizardry_classics.RunesofWizardry_Classics.preInit(RunesofWizardry_Classics.java:37)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:614)
	at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:252)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:230)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:147)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:604)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:98)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:331)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:122)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:483)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.util.ITooltipFlag
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 46 more
Caused by: java.lang.NullPointerException


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.12
	Operating System: Linux (amd64) version 3.14.32-xxxx-grs-ipv6-64
	Java Version: 1.8.0_112, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1433925632 bytes (1367 MB) / 2147483648 bytes (2048 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 4 total; -Xmx2048M -Xms2048M -XX:+UseG1GC -Xss2M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.40 Powered by Forge 14.21.1.2443 38 mods loaded, 38 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UCH	minecraft{1.12} [Minecraft] (minecraft.jar) 
	UCH	mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
	UCH	FML{8.0.99.99} [Forge Mod Loader] (custom.jar) 
	UCH	forge{14.21.1.2443} [Minecraft Forge] (custom.jar) 
	UCH	ivtoolkit{1.3.3-1.12} [IvToolkit] (minecraft.jar) 
	UCH	mantle{1.12-1.3.1.18} [Mantle] (Mantle-1.12-1.3.1.18.jar) 
	UCH	tconstruct{1.12-2.7.2.15} [Tinkers' Construct] (TConstruct-1.12-2.7.2.15.jar) 
	UCH	immersiveengineering{0.12-70} [Immersive Engineering] (ImmersiveEngineering-0.12-70.jar) 
	UCH	betterwithmods{1.2.9-1.11.2} [Better With Mods] (BetterWithMods-1.12-2.0.2.jar) 
	UCH	quark{r1.2-104} [Quark] (Quark-r1.2-104.jar) 
	UCH	autoreglib{1.3-13} [AutoRegLib] (AutoRegLib-1.3-13.jar) 
	UCH	baubles{1.5.1} [Baubles] (Baubles-1.12-1.5.1.jar) 
	UCH	betterwithaddons{@VERSION@} [Better with Addons] (Better With Addons-0.22.jar) 
	UCH	bibliocraft{2.4.3} [BiblioCraft] (BiblioCraft[v2.4.3][MC1.12.0].jar) 
	UCH	biomesoplenty{7.0.0.2291} [Biomes O' Plenty] (BiomesOPlenty-1.12-7.0.0.2291-universal.jar) 
	UCH	ceramics{1.12-1.3.2} [Ceramics] (Ceramics-1.12-1.3.2.jar) 
	UCH	chameleon{1.12-4.1.3} [Chameleon] (Chameleon-1.12-4.1.3.jar) 
	UCH	chiselsandbits{14.5} [Chisels & Bits] (chiselsandbits-14.5.jar) 
	UCH	concreteplus{1.1.2} [Concrete Extends] (ConcreteExtends-1.1.2_1.12.jar) 
	UCH	covens{0.1-dev} [Covens] (covens-0.1-dev.jar) 
	UCH	crafttweaker{4.0.2} [CraftTweaker2] (CraftTweaker2-1.12-4.0.2.jar) 
	UCH	cucumber{1.0.0} [Cucumber] (cucumber[1.12]-1.0.0.jar) 
	UCH	foamflower{1.12-0.0.1.0} [Foamflower] (foamflower-1.12-0.0.1.0.jar) 
	UCH	enderpay{1.12-0.0.5.1} [EnderPay] (enderpay-1.12-0.0.5.1.jar) 
	UCH	extraalchemy{0.3.4.1-release} [Extra Alchemy] (extraalchemy-0.3.4.1-release.jar) 
	UCH	gravestone{1.8.4} [Gravestone Mod] (gravestone-1.8.4.jar) 
	UCH	immersivepetroleum{1.1.3} [Immersive Petroleum] (immersivepetroleum-1.12-1.1.3.jar) 
	UCH	mysticalagriculture{1.6.2} [Mystical Agriculture] (mysticalagriculture[1.12]-1.6.2.jar) 
	UCH	mysticalagradditions{1.2.2} [Mystical Agradditions] (mysticalagradditions[1.12]-1.2.2.jar) 
	UCH	nex{2.1.11.12} [NetherEx] (NetherEx-1.12-2.1.11.12.jar) 
	UCH	plustic{6.0.0.5} [PlusTiC] (plustic-6.0.0.5.jar) 
	UCH	reccomplex{1.4.4} [Recurrent Complex] (RecurrentComplex-1.4.4.jar) 
	UCH	runesofwizardry{1.12-0.8.5-dev} [Runes of Wizardry] (Runes-of-Wizardry-master-1.12-0.8.5-dev.jar) 
	UCE	runesofwizardry_classics{1.12-0.5.2} [Runes of Wizardry - Classic Dusts Pack] (RunesofWizardry-Classics-1.12-0.5.2.jar) 
	UCH	storagedrawers{1.12-5.2.5} [Storage Drawers] (StorageDrawers-1.12-5.2.9.jar) 
	UCH	storagedrawersextra{@VERSION@} [Storage Drawers Extras] (StorageDrawersExtras-1.12-3.1.0.jar) 
	UCH	vending{1.12-2.0.0.0-beta2} [Vending] (vending-1.12-2.0.0.0-beta2.jar) 
	UCH	waystones{4.0.13} [Waystones] (Waystones_1.12.1-4.0.13.jar) 
	Loaded coremods (and transformers): 
LoadingPlugin (Quark-r1.2-104.jar)
  vazkii.quark.base.asm.ClassTransformer
IELoadingPlugin (ImmersiveEngineering-0.12-70.jar)
  blusunrize.immersiveengineering.common.asm.IEClassTransformer
IvToolkit (IvToolkit-1.3.3-1.12.jar)
  
Extra Alchemy ASM (extraalchemy-0.3.4.1-release.jar)
  zabi.minecraft.extraalchemy.asm.PotionTransformer
	Pulsar/tconstruct loaded Pulses: 
		- TinkerCommons (Enabled/Forced)
		- TinkerWorld (Enabled/Not Forced)
		- TinkerTools (Enabled/Not Forced)
		- TinkerHarvestTools (Enabled/Forced)
		- TinkerMeleeWeapons (Enabled/Forced)
		- TinkerRangedWeapons (Enabled/Forced)
		- TinkerModifiers (Enabled/Forced)
		- TinkerSmeltery (Enabled/Not Forced)
		- TinkerGadgets (Enabled/Not Forced)
		- TinkerOredict (Enabled/Forced)
		- TinkerIntegration (Enabled/Forced)
		- TinkerFluids (Enabled/Forced)
		- TinkerMaterials (Enabled/Forced)
		- TinkerModelRegister (Enabled/Forced)
		- chiselsandbitsIntegration (Enabled/Not Forced)

	Profiler Position: N/A (disabled)
	Is Modded: Definitely; Server brand changed to 'fml,forge'
	Type: Dedicated Server (map_server.txt)
```

I posted it in an answer in the Runes of Wizardry issue page too, but I thought this way it is easier to track, hop ethat is ok.

**Comment by Xilef11** [comment]
should be fixed in [v0.5.3](https://github.com/Xilef11/runesofwizardry-classics/releases/download/v1.12-0.5.3/RunesofWizardry-Classics-1.12-0.5.3.jar)

you *may* need to update the base mod to [0.8.5](https://github.com/Xilef11/Runes-of-Wizardry/releases/download/v1.12-0.8.5/Runes-of-Wizardry-1.12-0.8.5.jar) too

**Comment by Burgersim** [comment]
Again, thank you, I'll try it out right now ^^
